docs/cookbook/links.rst

```diff
@@ -0,0 +1,239 @@
+        type Movie {
+            required property title -> str;
+            required link director -> Person;
+            multi link cast -> Person;
+        }
+        type Person {
+            required property first_name -> str;
+            required property last_name -> str;
+        }
+    }
+
+In terms of SQL database we defined two foreign keys on the table containing
+movies. But link is much more than that.
+
+
+Nested Sets
+-----------
+
+Links allow fetching relatioships using single query:
+
+.. code-block:: edgeql-repl
+
+    tutorial> select Movie {
+    .........    director: { first_name },
+    .........    cast: { first_name },
+    ......... };
+    {
+        Object {
+            director: Object { first_name: 'Denis' },
+            cast: {
+                Object { first_name: 'Harrison' },
+                Object { first_name: 'Ryan' },
+                Object { first_name: 'Ana' },
+            }
+        }
+    }
+
+Aggregates
+----------
+
+Similarly you can run some aggregates on the nested sets:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Movie { title, cast_size:=count(.cast) };
+    {Object { title: 'Blade Runner 2049', cast_size: 3 }}
+
+You can find more information on aggregates in the
+:ref:`Cookbook <ref_cookbook_aggregates>` and the reference of
+:ref:`set functions <ref_eql_operators_set>` (or just search for "aggregate").
+
+
+.. _ref_cookbook_links_bw:
+
+Backward Links
+--------------
+
+In the movie example above, we have only shown a forward link traversal:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Movie { title, cast: { first_name } };
+    {
+        Object {
+            title: 'Blade Runner 2049',
+            cast: {
+                Object { first_name: 'Harrison' },
+                Object { first_name: 'Ryan' },
+                Object { first_name: 'Ana' },
+            }
+        }
+    }
+
+Here is a another example of using forward link. This time we only return
+last names of the artists as plain string (not an object). In this case, we
+need to alias the field with ``:=``. This example uses **forward link** ``.>``
+operator to make the code clearer:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Movie {
+    .........     title,
+    .........     starring := Movie.>cast.last_name,
+    ......... };
+    {Object {
+        title: 'Blade Runner 2049',
+        starring: {
+            'Ford',
+            'Gosling',
+            'de Armas',
+        }
+    }}
+
+To find all movies that a person is starred in we use a **backward link**
+``.<`` operator:
+
+.. code-block:: edgeql-repl
+
+    tutorial> SELECT Person {
+    .........     first_name,
+    .........     movies := Person.<cast[IS Movie].title,
+    ......... } FILTER .first_name = 'Ryan';
+    {Object {
+        first_name: 'Ryan',
+        movies: {
+            'Blade Runner 2049',
+        }
+    }}
+
+You might also note that we've added ``[IS Movie]`` cast. This is how backward
```

Comment:
> we've added ``[IS Movie]`` cast

We call this the "type filter" operator.  "Cast" implies no filtering, and a possibility of error being raised, whereas type filters always succeed (but may return an empty set),